src/pinning.js

```diff
@@ -88,10 +88,11 @@ class Pinning {
         if (onReplicatedFn) onReplicatedFn(address)
       })
     } else {
-      if (this.openDBs[address].dbPromise) {
-        await this.openDBs[address].dbPromise
+      if (!this.openDBs[address].dbPromise) {
+        // We don't need to call the responseFn if there is a promise present
+        // as it will get called anyway
+        responseFn(address)
```

Comment:
tricky, and this was mostly made worse by trying to open the same stores right? (which is resolved below), but even with that fix this is definitely more correct 👍

Reply:
Yeah this probably wasn't needed much when the duplicates got removed. But yeah, it's more correct :)